src/v/transform/logging/event.cc

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "transform/logging/event.h"
+
+#include "json/json.h"
+
+using namespace std::chrono_literals;
+
+namespace transform::logging {
```

Comment:
nit: Wrap all the internal stuff in the unnamed namespace to ensure internal linkage